masonry/src/passes/update.rs

```diff
@@ -401,3 +401,160 @@ pub(crate) fn run_update_anim_pass(root: &mut RenderRoot, elapsed_ns: u64) {
         elapsed_ns,
     );
 }
+
+// ----------------
+
+fn update_new_widgets(
+    global_state: &mut RenderRootState,
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    mut state: ArenaMut<'_, WidgetState>,
+) {
+    let _span = widget.item.make_trace_span().entered();
+
+    if !state.item.children_changed {
+        return;
+    }
+    state.item.children_changed = false;
+
+    // This will recursively call WidgetPod::lifecycle for all children of this widget,
```

Comment:
Is this not now only recursive for the "new" children?

Reply:
No, the point is that `Widget::lifecycle` is a "middleman" for calling `WidgetPod::lifecycle` on all children. This is also code that's going away soon, so it's fine if it's a little counter-intuitive.

Reply:
Oh, so the comment is just not right to say "recursive" at all?

Reply:
I mean, I guess that depends by what you mean by "recursively"? It's a single level of recursion.

Reply:
https://github.com/linebender/xilem/pull/589#discussion_r1761183070

---

If the callstack is:
```
update_new_widgets
Widget::lifecycle
WidgetPod::lifecycle
```
and never gets any deeper, that seems pretty non-recursive to me. Maybe I'm misunderstanding the code flows - sometimes things get confusing reviewing code with the +/- lines interleaved.